src/views/Package/PackageState.tsx

```diff
@@ -70,7 +70,7 @@ export const PackageStateProvider: FunctionComponent = ({ children }) => {
   const pkgName = scope ? `${scope}/${name}` : name;
   const allVersions = versionData?.packages[pkgName];
 
-  const markdown = useQuery(`${id}-docs-${language}`, () =>
+  const markdown = useQuery(`${id}-docs-${language}-${submodule}`, () =>
```

Comment:
All that was necessary to fix the core issue was updating this request id to take a dependency on the submodule